pkg/core/proxy/mockmanager.go

```diff
@@ -2,19 +2,34 @@ package proxy
 
 import (
 	"fmt"
+	"sort"
+	"strconv"
+	"strings"
 
 	"go.keploy.io/server/v2/pkg/models"
 )
 
+const (
+	filteredMock   = "filtered"
+	unFilteredMock = "unfiltered"
+	totalMock      = "total"
+)
+
 type MockManager struct {
 	filtered   *TreeDb
 	unfiltered *TreeDb
+	// usedMocks contains the name of the mocks as key which were used by the parsers during the test execution.
+	//
+	// value is an array that will contain the type of mock
+	usedMocks map[string][]string
```

Comment:
Used names as because the UpdateUnifilteredMocks/DeleteFilteredMocks are called multiple times so, to prevent addition of duplicate names into the usedMocks map. And this gives advatage to easily search that the mock with "mock-<?>" is used in O(1) instead of searching for the mock.name in the array to remove unused mocks. The filteredMock, unFilteredMock and totalMock are the constants only so they are defined at one place and used anywhere within the package.

Reply:
is totalMocks getting used anywhere..? I cant find any usage of it

Reply:
https://github.com/keploy/keploy/blob/f5ade3ca1360b094a7256356f3b5ee262e03ac16/pkg/core/proxy/mockmanager.go#L152